spec/realtime/failure.test.js

```diff
@@ -25,13 +25,14 @@ define(['ably', 'shared_helper', 'async'], function(Ably, helper, async) {
 	 * Connect with invalid credentials on various transports; connection state should be 'failed'
 	 */
 	exports.invalid_cred_failure = function(test) {
-		test.expect(availableTransports.length + 1);
+		test.expect((availableTransports.length + 1)*2);
 		try {
 			var failure_test = function(transports) {
 				return function(cb) {
 					var realtime = helper.AblyRealtime({key: "this.is:wrong", transports: transports});
 					realtime.connection.on('failed', function() {
 						test.ok(true, 'connection state for ' + transports + ' was failed, as expected');
+						test.notEqual(realtime.connection.errorReason, null, 'error reason was not set');
```

Comment:
Isn't this testing the wrong thing?  The spec says "and the Connection#errorReason will be set on the Connection object as well as the emitted ConnectionStateChange", so I would expect that you would add an argument to the function such as `realtime.connection.on('failed', function(connectionStateChange) {` and check that the error reason is present as well present on the errorReason attribute.  This test, as far as I can tell, proves that the spec is not curently met?